module/zfs/zfs_ioctl.c

```diff
@@ -5547,13 +5547,15 @@ dump_bytes(objset_t *os, void *buf, int len, void *arg)
 	dump_bytes_cb(&dbi);
 #else
 	/*
-	 * The vn_rdwr() call is performed in a taskq to ensure that there is
-	 * always enough stack space to write safely to the target filesystem.
-	 * The ZIO_TYPE_FREE threads are used because there can be a lot of
-	 * them and they are used in vdev_file.c for a similar purpose.
+	 * The zfs_file_write() call is performed in a taskq to ensure that
+	 * there is always enough stack space to write safely to the target
+	 * filesystem. system_delay_taskq so as not to block system_taskq if
+	 * write call blocks.
 	 */
-	spa_taskq_dispatch_sync(dmu_objset_spa(os), ZIO_TYPE_FREE,
-	    ZIO_TASKQ_ISSUE, dump_bytes_cb, &dbi, TQ_SLEEP);
+	taskqid id = taskq_dispatch(system_delay_taskq, dump_bytes_cb, &dbi,
+	    TQ_SLEEP);
```

Comment:
I suppose this write may be over network, and so stuck for a while, if not indefinitely.  There may be many of them same time, and I worry that they may block the taskq, that is used for other important tasks.  I am not saying ZIO_TYPE_FREE is the place for them, but may be we could create own taskq for each send?

Reply:
Systems that don't have `HAVE_LARGE_STACKS` should be quite uncommon these days.  This support dates back a decade to when 8K stacks were the default on x86_64 and 4K stacks were the default for i386.  Interestingly, I see there's finally a [proposal out there](https://lore.kernel.org/lkml/20240311164638.2015063-1-pasha.tatashin@soleen.com/) to allow dynamically sized stacks.

Reply:
@behlendorf interesting indeed, thanks for the pointer!